### PR TITLE
Milestone 3 - Users and Posts model Completed

### PR DESCRIPTION
**Pull request by Aaron-RN**
Summary:
- Added both User and Post model as well as the associations required between the two.
- Added Users and Posts controller which allows the creation of posts.
- Added a navbar, some styling and modified the views.

Hi TSE,

Can you please check our project and provide feedback on it?
Thank you in advance.

Kind regards,

Thaís and Aaron.

### CODE REVIEW

**Comment by thsvr** [comment]
> ## Changes required
> 
> Hi team!
> 
> Every requirement was almost meet, except for this:
> 
>     * [x]  Post is always displayed with content and author ( [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone3/posts4.png) from [Odin Project](https://www.theodinproject.com/courses/ruby-on-rails/lessons/final-project)) Below is the screen-shot of a post I created, only the content was displayed.
> 
> 
> Screenshot from 2020-01-09 01-27-58
> 
> After implementing the change submit another form for a code review.
> #### `Have great day` slightly_smiling_face
> 
> `Uchechukwu Henry Inyama`

Hi TSE,

we added the user's name to their posts as Uchechukwu required. 
Can you check if our project is able to be approved and provide feedback?

Thank you in advance :)

Kind regards,

Thaís and Aaron

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<h1> This is the index page </h1>
+
+<h2>All Posts</h2>
+<% @posts.each do |p|%>
+  <%= p.user.fname %> <%= p.user.lname %>
+  <br>
+  <%= p.content %>
+  <br><br>
+<% end %>
```
Comment (mcrd25) [inline app/views/users/index.html.erb]: This page really ought to display all users.
As seen from this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone3/userctr.png) from [odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/final-project)
But you are displaying all posts, please address.

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+class UsersController < ApplicationController
+  def index
+    @users = User.all
+    @posts = Post.all
```
Comment (mcrd25) [inline app/controllers/users_controller.rb]: Not sure this line is necessary for the `users#index` page.
[OPTIONAL]

**Comment by Aaron-RN** [comment]
> # Changes Required 🔍 👀
> Hi @Aaron-RN and @thsvr 👋 ,
> 
> You guys have done a great job with this so far! 👍
> There is only one hard requirement you have not met.
> 
> ### Required Changes
> * There ought to be a `users#index` page that lists all the app's users. See more in my review comment
> 
> ### Suggested changes [OPTIONAL]
> I am concerned that you currently have not implemented any tests and when I run `rspec` I get 10 errors!
> 
> See attached screenshot:
> Screen Shot 2020-01-09 at 5 52 05 PM
> 
> _**Please makes changes and submit another review request. If you have any questions, let me know here or on slack and I will be glad to clarify with you. You can also reach me and connect with me on my social media or email.**_
> 
> _Happy coding 😄,_
> **[--Maya](https://github.com/mcrd25)**
> [GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/) [GitLab](https://gitlab.com/mcrd25)
> 
> _P.S. all requested changes not marked by [OPTIONAL] or are not positive comments are **REQUIRED**_

Ok we will take the RSpec tests into consideration and will complete afterwards.
Thanks for your review.

**Pull request merged**